release/authorization/README.md

```diff
@@ -0,0 +1,419 @@
+denying subordinate portions of the permitted paths, or the converse, without
+regard to ordering of the rules in the configuration.
+
+## Best Match
+
+Authorization is performed for a singular user, gNMI path accessed, and access
+methodology (READ/WRITE). The result of an Authorization evaluation is an
+Action (Permit/Deny), policy version, and rule identifier.
+
+A Best, or most specific, match is that which has the longest match to the
+requested path and prefers:
+
+*   a specific user over a group in the matching policy.
+*   a defined KEY over a wildcard element in a keyed path.
+
+Authorization rules must be defined such that a single best match is possible.
+If the result of policy evaluation is more than one match, an error must be
+raised.
+
+Match rules permit a match against:
+
+*   User or Group (not both)
+*   an OpenConfig gNMI path
+*   an access method (READ or WRITE)
+
+An implicit deny is assumed, if there is no matching rule in the policy. Logging
+may be specified on a per-policy-rule basis as well as a default for the whole
+authorization policy.
+
+As a request is evaluated against the configured policy, a READ request for
+the configuration tree may traverse all of the tree. For portions of the tree
+for which the user has no access no data will be returned. A WRITE request
+which attempts to write to a denied subtree or element will return an error
+to the caller.
+
+[OpenConfig paths](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#222-paths)
+are heirarchical, and rooted at a defined "origin". OpenConfig may contain paths
+such as:
+
+```proto
+    /a/b/c/d
+```
+
+Paths may also have attributes associated with the path elements such as:
+
+```proto
+    /a/b[key=foo]/c/d
+```
+
+Attributes may be wildcarded in the policy, such as:
+
+```proto
+    /a/b[key=*]/c/d
+```
+
+Wildcards are only valid in policy rules when applied to attributes.
+Permitted use:
+
+```proto
+    /a/b[key=*]/c/d
+```
+
+Not permitted use:
+```proto
+    /a/b[key=foo]/*/d
+```
+
+An example of two rules with similar paths that differ only with respect
+to attribute wildcarding:
+
+```proto
+   /a/b[key=FOO]/c/d
+   /a/b[key=*]/c/d
+```
+
+The first rule specifies a single key, that rule is more specific than
+the second rule, which specifies 'any value is accepted'.
+
+
+## Bootstrap / Install Options
+
+System bootstrap, or install, operations may include an authorization policy
+delivered during bootstrap operations. It is suggested that the bootstrap
+process include the complete authorization policy so all production tools
+and services have immediate authorized access to finish installation and
+move devices into production in a timely manner.
+
+Using the Secure Zero Touch Provisioning (sZTP - RFC8572) process for
+bootstrap/installation provides a method to accomplish this delivery, and
+the delivery of all other bootstrap artifacts in a secure manner.
```

Comment:
Do we have something here that describes how this is to be done, or are we leaving this as a future exercise?

Reply:
I think this would fit into a bootstrap server implementation.

Effectively this is 1 more artifact to specify in the bootstrap process and delivery from the bootstrap server (or designated other endpoint/URL) to the device during bootstrapping.

Reply:
is using sZTP just a recommendation?  This section seems a bit outside the scope of the authorization model, though I don't have an issue with keeping it.

Reply:
Yes, I think carl/marcus were pushing for some recommendation to be included for:
  "Ok, so you want to do this, how would you do this for even the earliest part of a device life cycle?"
  
  if you want to do thigns then, you don't have ssh nor https/tls, if you want to deliver in a safe method it's either over console (ick) or the bootstrap process. The only 'safe' bootstrap for remote devices is .. sztp.

Reply:
done